src/validator.js

```diff
@@ -177,6 +177,37 @@ Validator.prototype = {
   },
 
   /**
+   * Extract value from nested object using string path with dot notation
+   *
+   * @param  {object} object to search in
+   * @param  {string} path inside object
+   * @return {any|void} value under the path
+   */
+  _objectPath: function (obj, path) {
+    if (obj.hasOwnProperty(path)) {
+      return obj[path];
+    }
+    
+    var keys = path.replace(/\[(\w+)\]/g, ".$1").replace(/^\./, "").split(".");
+    var copy = obj.constructor();
+
+    for (var attr in obj) {
```

Comment:
Why not use `Object.assign` here to copy the object? Or at least use a polyfill to support IE?